developer/src/common/web/utils/src/compiler-callbacks.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ *
+ * Callbacks for Keyman Developer compilers to ensure independence from
+ * filesystem, network, and console.
+ */
+
+import { CompilerEvent, CompilerCallbackOptions, CompilerErrorSeverity, CompilerError, CompilerMessageOverrideMap, CompilerErrorMask } from "./compiler-interfaces.js";
+
+/**
+ * Abstract interface for callbacks, to abstract out file i/o
+ */
+
+export interface CompilerCallbacks {
+  /**
+   * Attempt to load a file. Return falsy if not found.
+   * @param filename
+   */
+  loadFile(filename: string): Uint8Array;
+
+  /**
+   * Get file size, returns undefined if not found
+   */
+  fileSize(filename: string): number;
+
+  /**
+   * Returns true if file is a directory, undefined if not found
+   */
+  isDirectory(filename: string): boolean;
+
+  get path(): CompilerPathCallbacks;
+  get fs(): CompilerFileSystemCallbacks
+  get fsAsync(): CompilerFileSystemAsyncCallbacks;
+  get net(): CompilerNetAsyncCallbacks;
+
+  /**
+   * Resolves a file path relative to the baseFilename
+   * @param baseFilename
+   * @param filename
+   */
```

Comment:
How complete do we want to comment for these interfaces?
1. Do we want `@param filename` for some methods above?
2. What is `resolveFilename()` returning?

Reply:
Happy to chase this as a separate issue -- but it is low priority I think.